src/main/kotlin/org/jitsi/nlj/rtp/codec/vp9/Vp9Packet.kt

```diff
@@ -0,0 +1,128 @@
+        offset: Int,
+        length: Int
+    ) : this(buffer, offset, length,
+        isKeyframe = null,
+        isStartOfFrame = null,
+        encodingIndex = null,
+        height = null,
+        pictureId = null,
+        TL0PICIDX = null
+    )
+
+    /** Due to the format of the VP9 payload, this value is only reliable for packets where [isStartOfFrame] is true. */
+    override val isKeyframe: Boolean = isKeyframe ?: DePacketizer.VP9PayloadDescriptor.isKeyFrame(this.buffer, payloadOffset, payloadLength)
+
+    override val isStartOfFrame: Boolean = isStartOfFrame ?: DePacketizer.VP9PayloadDescriptor.isStartOfFrame(buffer, payloadOffset, payloadLength)
+
+    /** End of VP9 frame is the marker bit. */
+    /* TODO: frame/picture distinction here */
+    override val isEndOfFrame: Boolean
+        /** This uses [get] rather than initialization because [isMarked] is a var. */
+        get() = isMarked
+
+    val hasLayerIndices = DePacketizer.VP9PayloadDescriptor.hasLayerIndices(buffer, payloadOffset, payloadLength)
+
+    val hasPictureId = DePacketizer.VP9PayloadDescriptor.hasPictureId(buffer, payloadOffset, payloadLength)
+
+    val hasExtendedPictureId = DePacketizer.VP9PayloadDescriptor.hasExtendedPictureId(buffer, payloadOffset, payloadLength)
+
+    val hasTL0PICIDX = DePacketizer.VP9PayloadDescriptor.hasTL0PICIDX(buffer, payloadOffset, payloadLength)
+
+    var TL0PICIDX: Int by Delegates.observable(TL0PICIDX ?: DePacketizer.VP9PayloadDescriptor.getTL0PICIDX(buffer, payloadOffset, payloadLength)) {
+        _, _, newValue ->
+            if (newValue != -1 && !DePacketizer.VP9PayloadDescriptor.setTL0PICIDX(
+                    buffer, payloadOffset, payloadLength, newValue)) {
+                logger.cwarn { "Failed to set the TL0PICIDX of a VP9 packet." }
+            }
+        }
+
+    var pictureId: Int by Delegates.observable(pictureId ?: DePacketizer.VP9PayloadDescriptor.getPictureId(buffer, payloadOffset, payloadLength)) {
+        _, _, newValue ->
+            if (!DePacketizer.VP9PayloadDescriptor.setExtendedPictureId(
+                    buffer, payloadOffset, payloadLength, newValue)) {
+                logger.cwarn { "Failed to set the picture id of a VP9 packet." }
+            }
+        }
+
+    val temporalLayerIndex: Int = DePacketizer.VP9PayloadDescriptor.getTemporalLayerIndex(buffer, payloadOffset, payloadLength)
+
+    override var height: Int = height ?: -1
+
+    /**
+     * For [Vp9Packet] the payload excludes the VP9 Payload Descriptor.
+     */
+    override val payloadVerification: String
+        get() {
+            val rtpPayloadLength = payloadLength
+            val rtpPayloadOffset = payloadOffset
+            val VP9pdSize = DePacketizer.VP9PayloadDescriptor.getSize(buffer, rtpPayloadOffset, rtpPayloadLength)
+            val VP9PayloadLength = rtpPayloadLength - VP9pdSize
+            val hashCode = buffer.hashCodeOfSegment(payloadOffset + VP9pdSize, rtpPayloadOffset + rtpPayloadLength)
```

Comment:
Why not use the local `rtpPayloadOffset` instead of `payloadOffset` here?

Reply:
Also copied from VP8 - I can change it.